spanner/spansql/parser.go

```diff
@@ -2350,7 +2364,40 @@ func (p *parser) parseUnaryArithOp() (Expr, *parseError) {
 	if tok.err != nil {
 		return nil, tok.err
 	}
-	if op, ok := unaryArithOperators[tok.value]; ok {
+
+	op := tok.value
+
+	if op == "-" || op == "+" {
+		// If the next token is a numeric token, combine and parse as a literal.
+		ntok := p.next()
+		if ntok.err == nil {
+			switch ntok.typ {
+			case int64Token:
+				comb := op + ntok.value
+				n, err := strconv.ParseInt(comb, ntok.int64Base, 64)
+				if err != nil {
+					return nil, p.errorf("%v", err)
+				}
+				return IntegerLiteral(n), nil
+			case float64Token:
+				f := ntok.float64
+				if op == "-" {
+					f = -f
+				}
+				return FloatLiteral(f), nil
+			}
+		}
+		if op == "+" {
+			// A plus followed by anything that isn't a numeric literal is invalid.
+			// Spanner does not have unary plus.
```

Comment:
It seems to be undocumented, but the following is a valid query:

```
select +SingerId
from Singers
```

And so is also the following:

```
select ++-++SingerId
from Singers
```

So I would say it's ok to leave the implementation as is, but the comment should be changed.

Reply:
Well, let's just fix this now. It's a good point that we should match the real behaviour; this isn't the first time the docs have been wrong.

I've filed an internal bug (yet another!) about this, but I've also updated this change to properly handle unary plus.